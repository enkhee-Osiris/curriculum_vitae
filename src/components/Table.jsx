function Table({ data }) {
  return (
    <table className="w-full overflow-x-auto">
      <tbody className="divide-y divide-black dark:divide-gray-500">
        {Object.keys(data).map((key) => (
          <tr key={`table_${key.toLowerCase()}`}>
            <td className="py-2 font-medium">{key}</td>
            <td className="py-2">{data[key]}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

export default Table;
