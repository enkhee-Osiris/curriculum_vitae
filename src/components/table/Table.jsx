import React from 'react';
import { objectOf, string } from 'prop-types';

const Table = ({ datas }) => {
  const rows = Object.keys(datas).map(key => (
    <tr key={key}>
      <td style={{ fontWeight: 500 }}>{key}</td>
      <td>{datas[key]}</td>
    </tr>));

  return (
    <table className="u-full-width">
      <tbody>
        {rows}
      </tbody>
    </table>
  );
};

Table.propTypes = {
  datas: objectOf(string).isRequired,
};

export { Table as default };
