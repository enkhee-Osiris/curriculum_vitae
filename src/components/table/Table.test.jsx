import React from 'react';
import { shallow } from 'enzyme';
import Table from './Table';

describe('<Table />', () => {
  const props = {
    datas: { key: 'value' },
  };

  it('should be defined', () => {
    expect(Table).toBeDefined();
  });

  it('renders correctly', () => {
    const table = shallow(<Table {...props} />);

    expect(table).toMatchSnapshot();
  });

  it('renders tr', () => {
    const table = shallow(<Table {...props} />);

    expect(table.find('tr').length).toBeGreaterThan(0);
  });

  it('renders table datas', () => {
    const table = shallow(<Table {...props} />);

    expect(table.find('tr > td').first().text()).toEqual('key');
  });
});
