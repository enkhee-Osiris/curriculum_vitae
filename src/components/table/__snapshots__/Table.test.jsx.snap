// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    datas={
        Object {
            "key": "value",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <tbody>
        <tr>
                <td
                        style={
                                Object {
                                        "fontWeight": 500,
                                      }
                        }
                >
                        key
                </td>
                <td>
                        value
                </td>
        </tr>
</tbody>,
      "className": "u-full-width",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <tr>
            <td
                        style={
                                    Object {
                                                "fontWeight": 500,
                                              }
                        }
            >
                        key
            </td>
            <td>
                        value
            </td>
</tr>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "key",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <td
                style={
                                Object {
                                                "fontWeight": 500,
                                              }
                }
>
                key
</td>,
              <td>
                value
</td>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "key",
                "style": Object {
                  "fontWeight": 500,
                },
              },
              "ref": null,
              "rendered": "key",
              "type": "td",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "value",
              },
              "ref": null,
              "rendered": "value",
              "type": "td",
            },
          ],
          "type": "tr",
        },
      ],
      "type": "tbody",
    },
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tbody>
          <tr>
                    <td
                              style={
                                        Object {
                                                  "fontWeight": 500,
                                                }
                              }
                    >
                              key
                    </td>
                    <td>
                              value
                    </td>
          </tr>
</tbody>,
        "className": "u-full-width",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <td
                            style={
                                          Object {
                                                        "fontWeight": 500,
                                                      }
                            }
              >
                            key
              </td>
              <td>
                            value
              </td>
</tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "key",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  style={
                                    Object {
                                                      "fontWeight": 500,
                                                    }
                  }
>
                  key
</td>,
                <td>
                  value
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "key",
                  "style": Object {
                    "fontWeight": 500,
                  },
                },
                "ref": null,
                "rendered": "key",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "value",
                },
                "ref": null,
                "rendered": "value",
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
